src/main/java/com/google/devtools/build/remote/client/LogPrintingUtils.java

```diff
@@ -0,0 +1,144 @@
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.devtools.build.remote.client;
+
+import com.google.common.collect.Multiset;
+import com.google.common.collect.TreeMultiset;
+import com.google.devtools.build.lib.remote.logging.RemoteExecutionLog.LogEntry;
+import com.google.devtools.build.lib.remote.logging.RemoteExecutionLog.RpcCallDetails.DetailsCase;
+import com.google.devtools.build.lib.remote.logging.RemoteExecutionLog.WatchDetails;
+import com.google.devtools.build.remote.client.RemoteClientOptions.PrintLogCommand;
+import com.google.devtools.remoteexecution.v1test.ExecuteResponse;
+import com.google.longrunning.Operation;
+import com.google.longrunning.Operation.ResultCase;
+import com.google.protobuf.Timestamp;
+import com.google.watcher.v1.Change;
+import com.google.watcher.v1.Change.State;
+import com.google.watcher.v1.ChangeBatch;
+import io.grpc.Status.Code;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+
+/** Methods for printing log files. */
+public class LogPrintingUtils {
+  private static final String DELIMETER =
+      "---------------------------------------------------------\n";
+
+  /**
+   * Attempt to find and print ExecuteResponse from the details of a log entry for a Watch call. If
+   * no Operation could be found in the Watch call responses, or an Operation was found but failed,
+   * a failure message is printed.
+   */
+  private static void printExecuteResponse(WatchDetails watch) throws IOException {
+    for (ChangeBatch cb : watch.getResponsesList()) {
+      for (Change ch : cb.getChangesList()) {
+        if (ch.getState() != State.EXISTS) {
+          continue;
+        }
+        Operation o = ch.getData().unpack(Operation.class);
+        if (o.getResultCase() == ResultCase.ERROR && o.getError().getCode() != Code.OK.value()) {
+          System.out.printf("Error status in ChangeBatches: %s\n", o.getError().toString());
+          return;
+        } else if (o.getResultCase() == ResultCase.RESPONSE && o.getDone()) {
+          System.out.println("ExecuteResponse extracted:");
+          System.out.println(o.getResponse().unpack(ExecuteResponse.class).toString());
+          return;
+        }
+      }
+    }
+    System.out.println("Could not find ExecuteResponse in Watch call details.");
+  }
+
+  /** Print an individual log entry. */
+  private static void printLogEntry(LogEntry entry) throws IOException {
+    System.out.println(entry.toString());
+    if (entry.getDetails().getDetailsCase() == DetailsCase.WATCH) {
```

Comment:
I'm good with this cl, but I want to discuss this part a bit more.
(e.g., should we do this under some "summary" flag where we try to hide the "boring" stuff and only show what we think the user cares about, while leaving the "dump all readably" as an option with no smarter processing?)

Reply:
I'm okay with going either way. I think it's important we show ActionResults one way or another though.

Reply:
Ok, we talked offline about this.
The way Operation is currently printed, the ExecuteResponse is unreadable since it's packed inside an ANY. This line adds a readable version of it in well formed cases.
I'm good with keeping it this way for now